File: cpg-solidity/src/main/java/de/fraunhofer/aisec/cpg/frontends/solidity/nodes/UncheckedStatement.java
package de.fraunhofer.aisec.cpg.frontends.solidity.nodes;

import de.fraunhofer.aisec.cpg.graph.SubGraph;
import de.fraunhofer.aisec.cpg.graph.statements.Statement;
import org.jetbrains.annotations.Nullable;
import org.neo4j.ogm.annotation.Relationship;

public class UncheckedStatement extends Statement {
    @Relationship(value = "UNCHECKEDBLOCK", direction = "OUTGOING")
    @SubGraph("AST")
    @Nullable
    private Statement uncheckedBlock;

    @Nullable
    public final Statement getUncheckedBlock() {
        return this.uncheckedBlock;
    }

    public final void setUncheckedBlock(@Nullable Statement var1) {
        this.uncheckedBlock = var1;
        this.uncheckedBlock.addNextDFG(this);
    }

    public UncheckedStatement(@Nullable Statement uncheckedBlock) {
        this.uncheckedBlock = uncheckedBlock;
    }
}

File: cpg-solidity/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/solidity/nodes/UncheckedStatement.kt
package de.fraunhofer.aisec.cpg.frontends.solidity.nodes

import de.fraunhofer.aisec.cpg.graph.AST
import de.fraunhofer.aisec.cpg.graph.statements.Statement
import de.fraunhofer.aisec.cpg.graph.statements.expressions.Expression
import org.neo4j.ogm.annotation.Relationship

class UncheckedStatement() : Statement() {

    @field:Relationship(value = "UNCHECKEDBLOCK", direction = Relationship.Direction.OUTGOING)
    @AST
    var uncheckedBlock: Statement? = null
        set(var1) {
            field = var1
            uncheckedBlock!!.addNextDFG(this)
        }
}